### ISSUE TITLE
UnhandledPromiseRejectionWarning: TypeError: callback is not a function | TypeError: callback.apply is not a function

### ISSUE THREAD

**Issue by derbroti** [maxnowack/homebridge-eq3ble#23]
Sorry if this is in the end my fault, but even with your new 1.3.4 I now get:

```
(node:24694) UnhandledPromiseRejectionWarning: TypeError: callback is not a function
Feb 18 11:55:44 host homebridge[24694]: at /usr/local/lib/node_modules/homebridge-eq3ble/dist/callbackify.js:20:7
Feb 18 11:55:44 host homebridge[24694]: (node:24694) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
Feb 18 11:55:44 host homebridge[24694]: (node:24694) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
Feb 18 11:55:45 host homebridge[24694]: (node:24694) UnhandledPromiseRejectionWarning: TypeError: callback.apply is not a function
Feb 18 11:55:45 host homebridge[24694]: at /usr/local/lib/node_modules/homebridge-eq3ble/dist/callbackify.js:18:16
```
I am running this on node v11.9.0.
Which version do you develop this on?

Thanks for any input :)

**Comment by maxnowack** [comment]
Please try `v1.3.5`

**Comment by derbroti** [comment]
Sadly no :( I see the same issue still.

I can verify that I see your:
`if (callback) callback(err);` and  `if (callback) callback.apply(undefined, [null].concat(results));` in the build  dist/callbackify.js but still it complains with:
`(node:27159) UnhandledPromiseRejectionWarning: TypeError: callback.apply is not a function`

**Comment by maxnowack** [comment]
Can you put a `console.log(callback);` before both lines with `if (callback) …`?
(in `dist/callbackify.js`)

**Comment by derbroti** [comment]
I was just about to try that :)
```
Feb 18 12:35:09 host homebridge[27898]: [Function]
[...]
Feb 18 12:35:09 host homebridge[27898]: [Function]
Feb 18 12:35:09 host homebridge[27898]: 0
Feb 18 12:35:09 host homebridge[27898]: false
Feb 18 12:35:10 host homebridge[27898]: 10
```

Any idea where the 10 can come from? Home.App showes the current Temp as 10° and the Targettemp also. (Which seem to be default values?)

**Comment by maxnowack** [comment]
Can you change the `console.log` to `console.trace`? Maybe we see where the calls are coming from

**Comment by maxnowack** [comment]
There is a similar function in the [homebridge-tesla](https://github.com/nfarina/homebridge-tesla/blob/247b5584cc77b79e2aea6b48e265a60e27ad9a80/src/util/callbackify.ts) repository.
Please try to replace the contents of `dist/callbackify.js` with this:
````js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = callbackify;
// copied from https://github.com/nfarina/homebridge-tesla/blob/247b5584cc77b79e2aea6b48e265a60e27ad9a80/src/util/callbackify.ts
function callbackify(func) {
  return function () {
    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var onlyArgs = [];
    var maybeCallback = null;

    args.forEach(function (arg) {
      if (typeof maybeCallback === 'function') return;
      if (typeof arg === 'function') {
        maybeCallback = arg;
        return;
      }
      onlyArgs.push(arg);
    });

    if (!maybeCallback) {
      throw new Error('Missing callback parameter!');
    }

    var callback = maybeCallback;

    func.apply(undefined, onlyArgs).then(function (data) {
      return callback(null, data);
    }).catch(function (err) {
      return callback(err);
    });
  };
}
````

**Comment by derbroti** [comment]
Tracing didn't work - it just showed the callbackify line, nothing more?! I tried a catch(error) block with printing error.stack but that just froze everything?!

The replacement code you posted works flawlessly

**Comment by maxnowack** [comment]
Okay, give me a second. I'll release a new version with the replacement code :)

**Comment by maxnowack** [comment]
I've just released `v1.3.6` 😊
